### DISCUSSION TITLE
Problems understanding Extraction Rule files

### DISCUSSION THREAD

**Discussion by DTforwhat** [The-OpenROAD-Project/OpenROAD#6143] [Q&A] [answered]
After used OpenRCX tools  to generate rule files , I find it hard to understand the statistics in it , especially what do those OVERs and UNDERs stand for .
`xtraction Rules for OpenRCX

DIAGMODEL ON

LayerCount 10
DensityRate 1  0

DensityModel 0

Metal 1 RESOVER
WIDTH Table 1 entries:  0.09

Metal 1 RESOVER 0
DIST count 55 width 0.09
0 0 0 0.000955149
0 0.11 0 0.000955149
0 0.2 0 0.000955149
0 0.265 0 0.000955149
0 0.31 0 0.000955149
0 0.4 0 0.000955149
0 0.465 0 0.000955149
0 0.51 0 0.000955149
0 0.6 0 0.000955149
0 0.8 0 0.000955149
0.11 0.11 0 0.000955149
0.11 0.2 0 0.000955149
0.11 0.265 0 0.000955149
0.11 0.31 0 0.000955149
0.11 0.4 0 0.000955149
0.11 0.465 0 0.000955149
0.11 0.51 0 0.000955149
0.11 0.6 0 0.000955149
0.11 0.8 0 0.000955149
0.2 0.2 0 0.000955149
0.2 0.265 0 0.000955149
0.2 0.31 0 0.000955149
0.2 0.4 0 0.000955149
0.2 0.465 0 0.000955149
0.2 0.51 0 0.000955149
0.2 0.6 0 0.000955149
0.2 0.8 0 0.000955149
0.265 0.265 0 0.000955149
0.265 0.31 0 0.000955149
0.265 0.4 0 0.000955149
0.265 0.465 0 0.000955149
0.265 0.51 0 0.000955149
0.265 0.6 0 0.000955149
0.265 0.8 0 0.000955149
0.31 0.31 0 0.000955149
0.31 0.4 0 0.000955149
0.31 0.465 0 0.000955149
0.31 0.51 0 0.000955149
0.31 0.6 0 0.000955149
0.31 0.8 0 0.000955149
0.4 0.4 0 0.000955149
0.4 0.465 0 0.000955149
0.4 0.51 0 0.000955149
0.4 0.6 0 0.000955149
0.4 0.8 0 0.000955149
0.465 0.465 0 0.000955149
0.465 0.51 0 0.000955149
0.465 0.6 0 0.000955149
0.465 0.8 0 0.000955149
0.51 0.51 0 0.000955149
0.51 0.6 0 0.000955149
0.51 0.8 0 0.0000.6 0.6 0 0.000955149
0.6 0.8 0 0.000955149
0.8 0 0.8 0.000955149

Metal 1 OVER
WIDTH Table 1 entries:  0.09

Metal 1 OVER 0
DIST count 12 width 0.09
0.09 0 5.8371e-05 0.000477575
0.18 0 5.8371e-05 0.000477575
0.225 0 5.63576e-05 0.000477575
0.27 0 5.63576e-05 0.000477575
0.315 0 5.63576e-05 0.000477575
0.36 0 5.63576e-05 0.000477575
0.405 0 5.63576e-05 0.000477575
0.45 0 5.63576e-05 0.000477575
0.54 0 5.63576e-05 0.000477575
0.72 0 5.63576e-05 0.000477575
0.9 0 5.63576e-05 0.000477575
1.08 0 5.63576e-05 0.000477575
END DIST

Metal 1 UNDER
WIDTH Table 1 entries:  0.09

Metal 1 UNDER 2
DIST count 12 width 0.09
0.09 0 6.06993e-05 0.000477575
0.18 0 6.06993e-05 0.000477575
0.225 0 5.8371e-05 0.000477575
0.27 0 5.8371e-05 0.000477575
0.315 0 5.8371e-05 0.000477575
0.36 0 5.8371e-05 0.000477575
0.405 0 5.8371e-05 0.000477575
0.45 0 5.8371e-05 0.000477575
0.54 0 5.8371e-05 0.000477575
0.72 0 5.8371e-05 0.000477575
0.9 0 5.8371e-05 0.000477575
1.08 0 5.8371e-05 0.000477575
END DIST

Metal 1 UNDER 3
DIST count 12 width 0.09
0.09 0 6.18602e-05 0.000477575
0.18 0 6.18602e-05 0.000477575
0.225 0 6.27245e-05 0.000477575
0.27 0 6.27245e-05 0.000477575
0.315 0 6.34083e-05 0.000477575
0.36 0 6.34083e-05 0.000477575
0.405 0 6.40851e-05 0.000477575
0.45 0 6.40851e-05 0.000477575
0.54 0 6.52333e-05 0.000477575
0.72 0 6.52333e-05 0.000477575
0.9 0 6.21879e-05 0.000477575
1.08 0 6.49818e-05 0.000477575
END DIST

                                                      `

**Comment by maliberty** [comment, marked as answer]
They are models for a wire with metal on another layer over (above) or below (under).
